sympy/integrals/transforms.py

```diff
@@ -1799,15 +1844,110 @@ class LaplaceTransform(IntegralTransform):
 
     _name = 'Laplace'
 
-    def _compute_transform(self, f, t, s, **hints):
-        LT = _laplace_apply_rules(f, t, s, **hints)
-        if LT is None:
-            _simplify = hints.pop('simplify', True)
-            debug('_laplace_apply_rules could not match function %s'%(f,))
-            debug('    hints: %s'%(hints,))
-            return _laplace_transform(f, t, s, simplify=_simplify, **hints)
+    def _use_rules(self, **hints):
+        noconds = hints.get('noconds', False)
+        recursive = hints.get('recursive', 0)
+        hints['recursive'] = recursive+1
+        fn = self.function
+        t_ = self.function_variable
+        s_ = self.transform_variable
+        debug('[LT _u_r ] started with (%s, %s, %s)'%(fn, t_, s_))
+        debug('[LT _u_r ]     and hints %s'%(hints, ))
+        LT = None
+
+        def _is_uneval(LT):
+            decision = True
+            if LT.is_Add:
+                t = LT.args
+            else:
+                t = [LT]
+            for f in t:
+                if not f.has(LaplaceTransform):
+                    decision = False
+            return decision
+
+        # Try once without expanding
+        if recursive>0:
+            if recursive<=4:
+                r = expand(fn, deep=False)
+            else:
+                r = expand(fn)
+        else:
+            r = fn
+
+        if r.is_Add:
+            terms = r.args
+        else:
+            terms = [r]
+
+        by_rule = []
+        recursed = []
+        unevaluated = []
+        for f in terms:
+            LT = _laplace_apply_rules(f, t_, s_, **hints)
+            if LT is None:
+                if recursive>9:
+                    return None
+                LT = LaplaceTransform(f, t_, s_).doit(**hints)
+                if type(LT) is not tuple:
+                    if _is_uneval(LT):
+                        unevaluated.append(LT)
+                    else:
+                        recursed.append(LT)
+                else:
+                    if _is_uneval(LT[0]):
+                        unevaluated.append(LT[0])
+                    else:
+                        recursed.append(LT)
+            else:
+                if type(LT) is not tuple:
+                    if _is_uneval(LT):
+                        unevaluated.append(LT)
+                    else:
+                        if not noconds:
+                            unevaluated.append(LT)
+                        else:
+                            by_rule.append(LT)
+                else:
+                    if _is_uneval(LT[0]):
+                        unevaluated.append(LT[0])
+                    else:
+                        by_rule.append(LT)
```

Comment:
There's got to be cleaner way than all of these nested if/else. I think ideally the object passed around would always be a tuple. If `noconds` is set then the other two elements of the tuple can just be None.

Reply:
> I think ideally the object passed around would always be a tuple.

Very good idea, I think that will make things much simpler.

Reply:
OK, I have re-written all functions such that only tuples are passed around; the code is much simpler now. Will push in a minute ...